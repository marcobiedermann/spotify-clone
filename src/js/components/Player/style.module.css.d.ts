declare const styles: {
  readonly player: string;
  readonly player__controls: string;
};
export = styles;
