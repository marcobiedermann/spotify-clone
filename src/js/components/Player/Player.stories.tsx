import type { Meta, StoryObj } from '@storybook/react';
import Player from '.';

const meta: Meta<typeof Player> = {
  component: Player,
};

type Story = StoryObj<typeof Player>;

const Default: Story = {};

export { Default };
export default meta;
