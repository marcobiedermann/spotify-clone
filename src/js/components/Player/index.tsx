import React from 'react';
import { player, player__controls } from './style.module.css';

function Player(): JSX.Element {
  return (
    <div className={player}>
      <div>
        <div>Album Name</div>
        <div>Artist</div>
      </div>
      <div className={player__controls}>
        <button type="button">Shuffle</button>
        <div>
          <button type="button">Previous</button>
          <button type="button">Play</button>
          <button type="button">Pause</button>
          <button type="button">Next</button>
        </div>
        <button type="button">Repeat</button>
      </div>
      <div>
        <div>Volume</div>
      </div>
    </div>
  );
}

export default Player;
