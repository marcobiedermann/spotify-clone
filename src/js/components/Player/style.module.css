.player {
  background-color: #181818;
  border-top: 1px solid #282828;
  bottom: 0;
  display: flex;
  justify-content: space-between;
  padding: 1em;
  position: sticky;
}

.player__controls {
  display: flex;
}
