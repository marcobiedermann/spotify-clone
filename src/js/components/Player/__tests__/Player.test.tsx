import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import * as stories from '../Player.stories';

const { Default } = composeStories(stories);

describe('Player component', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
