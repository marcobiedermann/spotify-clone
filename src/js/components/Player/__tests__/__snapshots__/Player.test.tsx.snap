// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Player component > should render correctly 1`] = `
<div>
  <div>
    <div>
      Album Name
    </div>
    <div>
      Artist
    </div>
  </div>
  <div>
    <button
      type="button"
    >
      Shuffle
    </button>
    <div>
      <button
        type="button"
      >
        Previous
      </button>
      <button
        type="button"
      >
        Play
      </button>
      <button
        type="button"
      >
        Pause
      </button>
      <button
        type="button"
      >
        Next
      </button>
    </div>
    <button
      type="button"
    >
      Repeat
    </button>
  </div>
  <div>
    <div>
      Volume
    </div>
  </div>
</div>
`;
